lib/Sema/TypeChecker.h

```diff
@@ -924,7 +924,8 @@ enum class UnsupportedMemberTypeAccessKind : uint8_t {
 /// member of the given base type.
 UnsupportedMemberTypeAccessKind
 isUnsupportedMemberTypeAccess(Type type, TypeDecl *typeDecl,
-                              bool hasUnboundOpener);
+                              bool hasUnboundOpener,
+                              bool isExtensionBinding = false);
```

Comment:
Instead of passing two booleans, maybe it's better to pass the TypeResolverOptions?

Reply:
Do you mean a tuple of booleans?

Reply:
This is a good idea; however, in `resolveNestedIdentTypeComponent` function, we are retrieving `hasUnboundOpener`boolean value from `TypeResolution`, not from its options. So I don't think passing options here would b enough. As an alternative, though, I can think of a way of somehow combining these two flags into one.